Automattic/wp-calypso
# Gutenboarding: update site creation flow

#### Design and functional updates:
* Add Domains step and render Domains Modal as a Step.
* Render Plans Modal as a Step.
* If the user enters a site title on Intent Capture then show Domain step next, before Design Selection.
* If the user skips entering a site title, then show Domain step after Style Preview (before Plans).
* Remove Continue button from Plans Page and submit on Select.
* Remove pre-selected plan from Plans page.
* Hide PlansButton and DomainPickerButton from Intent Capture, Domains and Plans steps.
* Align intent capture CTA to right on mobile
* Change copy for paid domains price label.
* Add max-width to Gutenboarding except design selection and create-site.

#### Technical updates:
* Add useStepNavigation hook and use it in Gutenboarding steps.
* Add BackButton, NextButton and SkipButton to ActionButtons.
* Use only ActionButtons in Gutenboarding for consistency.
* Trigger navigation to previous and next steps in useStepNavigation hook.
* Add useOnLogin and useSignup hooks and extract logic from header.
* Update abtest slug
* Fix Gutenboarding e2e tests

####  Analytics events update
  -  `calypso_newsite_modal_open` with prop `{ name: 'PlansGrid' }` becomes `calypso_newsite_step_enter` with prop `{ step: 'PlansModal' }`.
  -  `calypso_newsite_modal_close` with prop `{ name: 'PlansGrid' }` becomes `calypso_newsite_step_leave` with prop `{ step: 'PlansModal' }`.

#### Follow up:
- Register new analytics events and changes: https://github.com/Automattic/wp-calypso/issues/42250

#### Testing instructions

* Go to [/new](https://calypso.live/new?branch=update/gutenboarding-site-creation-flow)
* Test whole Gutenboarding flow and check for the changes mentioned above.
* Check analytics event to fire as before with the changes mentioned above.
* Check different routes including i18n and plan slugs (eg: `/new/design/business/es`).

Fixes https://github.com/Automattic/wp-calypso/issues/43134 and https://github.com/Automattic/wp-calypso/issues/43133
Fixes #43205
